typings/dom-tag-component-factories.ts

```diff
@@ -0,0 +1,141 @@
+  DomTagComponentFactory<"map", Theme>,
+  DomTagComponentFactory<"mark", Theme>,
+  DomTagComponentFactory<"menu", Theme>,
+  DomTagComponentFactory<"menuitem", Theme>,
+  DomTagComponentFactory<"meta", Theme>,
+  DomTagComponentFactory<"meter", Theme>,
+  DomTagComponentFactory<"nav", Theme>,
+  DomTagComponentFactory<"noscript", Theme>,
+  DomTagComponentFactory<"object", Theme>,
+  DomTagComponentFactory<"ol", Theme>,
+  DomTagComponentFactory<"optgroup", Theme>,
+  DomTagComponentFactory<"option", Theme>,
+  DomTagComponentFactory<"output", Theme>,
+  DomTagComponentFactory<"p", Theme>,
+  DomTagComponentFactory<"param", Theme>,
+  DomTagComponentFactory<"picture", Theme>,
+  DomTagComponentFactory<"pre", Theme>,
+  DomTagComponentFactory<"progress", Theme>,
+  DomTagComponentFactory<"q", Theme>,
+  DomTagComponentFactory<"rp", Theme>,
+  DomTagComponentFactory<"rt", Theme>,
+  DomTagComponentFactory<"ruby", Theme>,
+  DomTagComponentFactory<"s", Theme>,
+  DomTagComponentFactory<"samp", Theme>,
+  DomTagComponentFactory<"script", Theme>,
+  DomTagComponentFactory<"section", Theme>,
+  DomTagComponentFactory<"select", Theme>,
+  DomTagComponentFactory<"small", Theme>,
+  DomTagComponentFactory<"source", Theme>,
+  DomTagComponentFactory<"span", Theme>,
+  DomTagComponentFactory<"strong", Theme>,
+  DomTagComponentFactory<"style", Theme>,
+  DomTagComponentFactory<"sub", Theme>,
+  DomTagComponentFactory<"summary", Theme>,
+  DomTagComponentFactory<"sup", Theme>,
+  DomTagComponentFactory<"table", Theme>,
+  DomTagComponentFactory<"tbody", Theme>,
+  DomTagComponentFactory<"td", Theme>,
+  DomTagComponentFactory<"textarea", Theme>,
+  DomTagComponentFactory<"tfoot", Theme>,
+  DomTagComponentFactory<"th", Theme>,
+  DomTagComponentFactory<"thead", Theme>,
+  DomTagComponentFactory<"time", Theme>,
+  DomTagComponentFactory<"title", Theme>,
+  DomTagComponentFactory<"tr", Theme>,
+  DomTagComponentFactory<"track", Theme>,
+  DomTagComponentFactory<"u", Theme>,
+  DomTagComponentFactory<"ul", Theme>,
+  DomTagComponentFactory<"var", Theme>,
+  DomTagComponentFactory<"video", Theme>,
+  DomTagComponentFactory<"wbr", Theme>,
+  DomTagComponentFactory<"circle", Theme>,
+  DomTagComponentFactory<"clipPath", Theme>,
+  DomTagComponentFactory<"defs", Theme>,
+  DomTagComponentFactory<"ellipse", Theme>,
+  DomTagComponentFactory<"g", Theme>,
+  DomTagComponentFactory<"image", Theme>,
+  DomTagComponentFactory<"line", Theme>,
+  DomTagComponentFactory<"linearGradient", Theme>,
+  DomTagComponentFactory<"mask", Theme>,
+  DomTagComponentFactory<"path", Theme>,
+  DomTagComponentFactory<"pattern", Theme>,
+  DomTagComponentFactory<"polygon", Theme>,
+  DomTagComponentFactory<"polyline", Theme>,
+  DomTagComponentFactory<"radialGradient", Theme>,
+  DomTagComponentFactory<"rect", Theme>,
+  DomTagComponentFactory<"stop", Theme>,
+  DomTagComponentFactory<"svg", Theme>,
+  DomTagComponentFactory<"text", Theme>,
+  DomTagComponentFactory<"tspan", Theme> {}
```

Comment:
Sorry, that is really something wrong with that approach if you had to list all the tags in that way.
You probably need to use mapped types along with string union types.

Reply:
I'd love if someone could work out a way to achieve it with mapped types and string union types.

Afaict mapped types don't work with function arguments.